context.go

```diff
@@ -24,9 +28,21 @@ type Context struct {
 // NewContext creates a new context. For use in when invoking an App or Command action.
 func NewContext(app *App, set *flag.FlagSet, parentCtx *Context) *Context {
 	c := &Context{App: app, flagSet: set, parentContext: parentCtx}
-
 	if parentCtx != nil {
+		if parentCtx.Context != nil {
+			parentCtx.Context = context.Background()
+		}
```

Comment:
This has a side effect on the parentCtx passed in, which I as a user would find suprising.

Reply:
Also, I think the intended condition in the if was == nil it seems?

I think this needs tests, as we can't rely on reviewers to prove this works

Reply:
@AudriusButkevicius this is just to make sure the context is never nil, so that we never panic. Should I remove it and leave the risk up to the user?

Reply:
It's not true, the if statement is wrong and it always overrides it if its *NOT* nil, and leaves it nil otherwise. I think we should not cause side effects on objects given to us, so if its nil, create a new one locally, and then validate this behaviour with a few tests.

Reply:
@AudriusButkevicius great catch, that was a mistake on my part. I will change != nil to == nil

Reply:
@AudriusButkevicius i also added some propagation tests to ensure the parent context does not get overridden.

Reply:
I'd like to add sigterm tests, but i'm not sure how I can send a sigterm signal to the process, without killing the tests themselves.

Reply:
I still think we should not mutate the parent context.
```
c.Context = parentCtx.Context
```

Then outside the if (oppose to else):
```
If c.Context == nil {
  // Create cancel context, spin up signal handler ctx.
}
```

Reply:
@AudriusButkevicius done